Assets/Scripts/WorldProperties.cs

```diff
@@ -91,6 +93,111 @@ void Start()
             clipShader = GameObject.FindWithTag("Ground").GetComponent<Renderer>().material.shader;
         }
 
+        /// <summary>
+        /// Cycle through the connected drones
+        /// </summary>
+        public static void SelectNextDrone()
+        {
+            Debug.Log("Selection next drone");
+
+            if(selectedDrone != null)
+            {
+                dronesQueue.Enqueue(selectedDrone);
+            }
+
+            if (dronesQueue.Count > 0)
+            {
+                Drone nextDrone = dronesQueue.Dequeue();
+                nextDrone.gameObjectPointer.GetComponent<DroneProperties>().SelectDrone();
+            }
+            else
+            {
+                Debug.Log("No drones connected");
+            }
+
+        }
+
+        /// <summary> 
+        /// Get nextDroneID 
+        /// </summary>
+        public static char GetNextDroneID()
+        {
+            return nextDroneId;
+        }
+        
+        /// <summary>
+        /// Increment the next drone ID
+        /// </summary>
+        public static void IncrementDroneID()
+        {
+            nextDroneId++;
+        }
+
+        /// <summary>
+        /// Update the selected drone.
+        /// </summary>
+        public static void UpdateSelectedDrone(Drone newSelectedDrone)
+        {
+            if (selectedDrone != null)
+            {
+                selectedDrone.gameObjectPointer.GetComponent<DroneProperties>().DeselectDrone();
```

Comment:
When you run DeselectDrone(), does it get put back in the queue?

Reply:
yep